dissect/target/plugins/child/hyperv.py

```diff
@@ -43,16 +43,17 @@ def list_children(self) -> Iterator[ChildTargetRecord]:
         if self.data_vmcx.exists():
             data = hyperv.HyperVFile(self.data_vmcx.open()).as_dict()
 
-            for vm_path in data["Configurations"]["VirtualMachines"].values():
-                yield ChildTargetRecord(
-                    type=self.__type__,
-                    path=path.from_windows(vm_path),
-                    _target=self.target,
-                )
+            if virtual_machines := data["Configurations"].get("VirtualMachines"):
+                for vm_path in virtual_machines.values():
+                    yield ChildTargetRecord(
+                        type=self.__type__,
+                        path=path.from_windows(vm_path),
+                        _target=self.target,
+                    )
 
         for xml_path in self.vm_xml:
             yield ChildTargetRecord(
                 type=self.__type__,
-                path=xml_path.resolve(),
+                path=path.from_windows(xml_path.resolve()),
```

Comment:
I don't believe this to be necessary, at least not with the latest changes in `flow.record`.

Reply:
Using `flow.record==3.10` removing the `path.from_windows` call causes the test to fail:

```
>       assert (
            str(children[4].path)
            == "\\sysvol\\ProgramData\\Microsoft\\Windows\\Hyper-V\\Virtual Machines\\B90AC31B-C6F8-479F-9B91-07B894A6A3F6.xml"  # noqa E501
        )
E       AssertionError: assert '/sysvol/Prog...894A6A3F6.xml' == '\\sysvol\\Pr...894A6A3F6.xml'
E         - \sysvol\ProgramData\Microsoft\Windows\Hyper-V\Virtual Machines\B90AC31B-C6F8-479F-9B91-07B894A6A3F6.xml
E         ? ^      ^           ^         ^       ^       ^                ^
E         + /sysvol/ProgramData/Microsoft/Windows/Hyper-V/Virtual Machines/B90AC31B-C6F8-479F-9B91-07B894A6A3F6.xml
E         ? ^      ^           ^         ^       ^       ^                ^

tests/test_plugins_child_hyperv.py:35: AssertionError
```

Reply:
Hm I'm confused now on the expected behaviour of this 😄 @pyrco can you jump in?

Reply:
That's interesting, I did some testing and debugging and it behaves as expected with flow.record 3.10 on my system and on the GitHub test runners, no need for `.from_windows()` here.

The behavior you describe is what you would get if you had a flow.record <= 3.9. I changed my tox py3 venv and explicitly installed flow.record==3.9 and saw the exact behavior you describe.

Perhaps tox did not detect properly it should update its py3 venv, this happens sometimes. If you could rerun the tests with `tox -re py3` and see if that changes anything. The `-r` will force the recreation of tox' py3 test venv.

Reply:
Thanks @pyrco. It seems tox's aggressive caching was the culprit here indeed.